core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/DataAccessImpl.java

```diff
@@ -1046,39 +1310,14 @@ private BoundStatement bindDataPoint(PreparedStatement statement, Metric<?> metr
     }
 
     @Override
-    public Observable<Row> findAvailabilityData(MetricId<AvailabilityType> id, long startTime, long endTime,
-            int limit, Order order, int pageSize) {
-        if (order == Order.ASC) {
-            if (limit <= 0) {
-                return rxSession.executeAndFetch(findAvailabilitiesASC.bind(id.getTenantId(),
-                        AVAILABILITY.getCode(), id.getName(), DPART, getTimeUUID(startTime), getTimeUUID(endTime))
-                        .setFetchSize(pageSize));
-            } else {
-                return rxSession.executeAndFetch(findAvailabilitiesWithLimitASC.bind(id.getTenantId(),
-                        AVAILABILITY.getCode(), id.getName(), DPART, getTimeUUID(startTime), getTimeUUID(endTime),
-                        limit).setFetchSize(pageSize));
-            }
-        } else {
-            if (limit <= 0) {
-                return rxSession.executeAndFetch(findAvailabilities.bind(id.getTenantId(), AVAILABILITY.getCode(),
-                        id.getName(), DPART, getTimeUUID(startTime), getTimeUUID(endTime)).setFetchSize(pageSize));
-            } else {
-                return rxSession.executeAndFetch(findAvailabilitiesWithLimit.bind(id.getTenantId(),
-                        AVAILABILITY.getCode(), id.getName(), DPART, getTimeUUID(startTime), getTimeUUID(endTime),
-                        limit).setFetchSize(pageSize));
-            }
+    public <T> Observable<ResultSet> deleteMetricData(MetricId<T> id) {
+        if(id.getType() == STRING) {
+            return rxSession.execute(deleteMetricData.bind(id.getTenantId(), id.getType().getCode(), id.getName(), DPART));
         }
-    }
-
-    @Override
-    public Observable<Row> findAvailabilityData(MetricId<AvailabilityType> id, long timestamp) {
-        return rxSession.executeAndFetch(findAvailabilityByDateRangeInclusive.bind(id.getTenantId(),
-                AVAILABILITY.getCode(), id.getName(), DPART, UUIDs.startOf(timestamp), UUIDs.endOf(timestamp)));
-    }
 
-    @Override
-    public <T> Observable<ResultSet> deleteMetricData(MetricId<T> id) {
-        return rxSession.execute(deleteMetricData.bind(id.getTenantId(), id.getType().getCode(), id.getName(), DPART));
+        return getPrepForAllTempTables(TempStatement.DELETE_DATA)
```

Comment:
If we don't delete the data, then the compress job will reinsert them to the data_compressed and then the jobs "DeleteTenant" or any other delete don't really do much anymore. As there's no place they would delete from.